--- i/include/tbb/tbb_config.h
+++ w/include/tbb/tbb_config.h
@@ -17,6 +17,11 @@
 #ifndef __TBB_tbb_config_H
 #define __TBB_tbb_config_H
 
+// ports config begin
+#define TBB_USE_EXCEPTIONS 0
+#define __TBB_NO_IMPLICIT_LINKAGE 1
+// ports config end
+
 /** This header is supposed to contain macro definitions and C style comments only.
     The macros defined here are intended to control such aspects of TBB build as
     - presence of compiler features
