--- i/doctest/doctest.h
+++ w/doctest/doctest.h
@@ -42,6 +42,9 @@
 #ifndef DOCTEST_LIBRARY_INCLUDED
 #define DOCTEST_LIBRARY_INCLUDED
 
+#define DOCTEST_CONFIG_IMPLEMENTATION_IN_DLL
+#define DOCTEST_CONFIG_USE_IOSFWD
+
 // =================================================================================================
 // == VERSION ======================================================================================
 // =================================================================================================
