--- i/include/date/date.h
+++ w/include/date/date.h
@@ -32,6 +32,10 @@
 // been invented (that would involve another several millennia of evolution).
 // We did not mean to shout.
 
+// ports config begin
+#define ONLY_C_LOCALE 1
+// ports config end
+
 #ifndef HAS_STRING_VIEW
 #  if __cplusplus >= 201703 || (defined(_MSVC_LANG) && _MSVC_LANG >= 201703L)
 #    define HAS_STRING_VIEW 1
--- i/include/date/tz.h
+++ w/include/date/tz.h
@@ -43,6 +43,21 @@
 // required. On Windows, the names are never "Standard" so mapping is always required.
 // Technically any OS may use the mapping process but currently only Windows does use it.
 
+// ports config begin
+#ifndef NDEBUG
+#define DATE_USE_DLL 1
+#endif
+
+#ifdef _WIN32
+#define USE_OS_TZDB 0
+#else
+#define USE_OS_TZDB 1
+#endif
+
+#define HAS_REMOTE_API 0
+// ports config end
+
+
 #ifndef USE_OS_TZDB
 #  define USE_OS_TZDB 0
 #endif
