--- i/libtiff/tif_stream.cxx
+++ w/libtiff/tif_stream.cxx
@@ -25,6 +25,7 @@
 /*
  * TIFF Library UNIX-specific Routines.
  */
+#include "tiffio.hxx"
 #include "tiffiop.h"
 #include <iostream>
 
--- i/libtiff/tiffio.hxx
+++ w/libtiff/tiffio.hxx
@@ -25,16 +25,27 @@
 #ifndef _TIFFIO_HXX_
 #define	_TIFFIO_HXX_
 
+#if defined(_WIN32) && defined(ACE_SHARED)
+#  ifdef TIFF_EXPORT
+#    define TIFF_API __declspec(dllexport)
+#  else
+#    define TIFF_API __declspec(dllimport)
+#  endif
+#else
+#  define TIFF_API
+#endif
+
 /*
  * TIFF I/O library definitions which provide C++ streams API.
  */
 
-#include <iostream>
+#include <istream>
+#include <ostream>
 #include "tiff.h"
 #include "tiffio.h"
 
-extern TIFF* TIFFStreamOpen(const char*, std::ostream *);
-extern TIFF* TIFFStreamOpen(const char*, std::istream *);
+TIFF_API TIFF* TIFFStreamOpen(const char*, std::ostream *);
+TIFF_API TIFF* TIFFStreamOpen(const char*, std::istream *);
 
 #endif /* _TIFFIO_HXX_ */
 
